Chiller master JSON with heat rejection TCDs

jugonzal07 opened the pull request:
Updated chiller master JSON with heat rejection TCDs for Section 22 rules12,13, 15,17,18

yunjoonjung-PNNL commented:
Juan, Could you address the following comments?

22-15  
1) ```loop``` value under the ```heat_rejections``` should be ```Condenser Loop 1``` instead of ```Chiller Loop 1```
2) ```approach``` key under the ```heat_rejections``` is missing

22-17
1) ```loop``` value under the ```heat_rejections``` should be ```Condenser Loop 1``` instead of ```Chiller Loop 1```
(minor) ```primary_rule``` key should be ```No``` instead of ```Yes```
(b) - expected value should be ```undetermined``` instead of ```pass```

22-18
1) ```loop``` value under the ```heat_rejections``` should be ```Condenser Loop 1``` instead of ```Chiller Loop 1```

Other than these, it looks great to me! Thanks Juan.

jugonzal07 commented:
> Juan, Could you address the following comments?
> 
> 22-15
> 
> 1. `loop` value under the `heat_rejections` should be `Condenser Loop 1` instead of `Chiller Loop 1`
> 2. `approach` key under the `heat_rejections` is missing
> 
> 22-17
> 
> 1. `loop` value under the `heat_rejections` should be `Condenser Loop 1` instead of `Chiller Loop 1`
>    (minor) `primary_rule` key should be `No` instead of `Yes`
>    (b) - expected value should be `undetermined` instead of `pass`
> 
> 22-18
> 
> 1. `loop` value under the `heat_rejections` should be `Condenser Loop 1` instead of `Chiller Loop 1`
> 
> Other than these, it looks great to me! Thanks Juan.

Thanks Yun Joon. I've pushed those updates for you. Let me know if you mind that I put it all together in this branch. It just seems easier to do that rather than manage so many PRs. Let me know if you disagree and I can split them up.

yunjoonjung-PNNL commented on rct229/ruletest_engine/ruletest_jsons/ashrae9012019/section22/rule_22_15.json:
```diff
@@ -0,0 +1,464 @@
+                                "loop": "Condenser Loop 1",
+                                "design_wetbulb_temperature": 15.5555555555556,
+                                "approach": 6.28888888888889
+                            }
```
Juan, this is a bit tricky. In the schedule, the ```design_wetbulb_temperature``` is in degC and ```approach``` is in K.
And, rule 22-15 compares whether the two temperatures are the same or not. Thus, I believe approach temp should be 288 K. Could you check this out?

yunjoonjung-PNNL commented:
I tested out rule 22 - 35, 37, 38, and 39 with TCDs in this PR on the develop branch. All rules passed as expected!

yunjoonjung-PNNL commented:
22-41 passed as expected.

Questions to @jugonzal07 )
The 22-16 TCD in this PR failed and original TCD passed. Did you make the 22-16 changes unexpectedly?

weilixu commented:
> I tested out rule 22 - 35, 37, 38, and 39 with TCDs in this PR on the develop branch. All rules passed as expected!

I do not see 35, 37, 38 and 39 test json in this PR. could you generate them again?

jugonzal07 commented:
> > I tested out rule 22 - 35, 37, 38, and 39 with TCDs in this PR on the develop branch. All rules passed as expected!
> 
> I do not see 35, 37, 38 and 39 test json in this PR. could you generate them again?

Thanks Weili, I mistakenly used the wrong spreadsheet when swapping to my new machine. I've addressed this and updated this with all missing TCDs. I contacted @yunjoonjung-PNNL to look at what's happening with 22-16 and 22-19. I made updates to ensures cooling towers had associated condensing loops and it seems to have caused a few issues with those rules.

weilixu commented:
> > > I tested out rule 22 - 35, 37, 38, and 39 with TCDs in this PR on the develop branch. All rules passed as expected!
> > 
> > 
> > I do not see 35, 37, 38 and 39 test json in this PR. could you generate them again?
> 
> Thanks Weili, I mistakenly used the wrong spreadsheet when swapping to my new machine. I've addressed this and updated this with all missing TCDs. I contacted @yunjoonjung-PNNL to look at what's happening with 22-16 and 22-19. I made updates to ensures cooling towers had associated condensing loops and it seems to have caused a few issues with those rules.

Thank you Juan, 22-16 and 19 seems not in the scope. These are primary rules?

jugonzal07 commented:
> > > > I tested out rule 22 - 35, 37, 38, and 39 with TCDs in this PR on the develop branch. All rules passed as expected!
> > > 
> > > 
> > > I do not see 35, 37, 38 and 39 test json in this PR. could you generate them again?
> > 
> > 
> > Thanks Weili, I mistakenly used the wrong spreadsheet when swapping to my new machine. I've addressed this and updated this with all missing TCDs. I contacted @yunjoonjung-PNNL to look at what's happening with 22-16 and 22-19. I made updates to ensures cooling towers had associated condensing loops and it seems to have caused a few issues with those rules.
> 
> Thank you Juan, 22-16 and 19 seems not in the scope. These are primary rules?

Correct-- I noticed this issue across other heat rejection unit test where cooling towers were incorrectly linked to the Chiller Loops rather than the Chillers being correctly linked to the Condensing Loop. I suppose this doesn't quite fit the scope for non-primary rules, but I thought it worthwhile to fix while we made these fixes. 

If you'd like, I can undo this change for 16 and 19 and make a separate PR for those.

yunjoonjung-PNNL commented:
Juan, I just tested out 22-16,19. Could you see if the below-suggesting changes make sense to you?

For both 22-16 and 19 rules, the ```cooling_or_condensing_design_and_control``` key under the ```Chiller Loop 1``` loop should be moved to the ```Condenser Loop 1``` loop.

After changing this, all 16 and 19 testing cases passed!

jugonzal07 commented:
> Juan, I just tested out 22-16,19. Could you see if the below-suggesting changes make sense to you?
> 
> For both 22-16 and 19 rules, the `cooling_or_condensing_design_and_control` key under the `Chiller Loop 1` loop should be moved to the `Condenser Loop 1` loop in the `fluid_loops`.
> 
> After changing this, all 16 and 19 testing cases passed!

Thanks Yun Joon! That fixed it.

weilixu commented:
Thank you everyone, I will merge this PR.
